core/src/test/scala/integration/kafka/api/ConsumerTopicCreationTest.scala

```diff
@@ -17,49 +17,58 @@
 
 package kafka.api
 
-import java.lang.{Boolean => JBoolean}
-import java.time.Duration
-import java.util
-import java.util.Collections
-
 import kafka.server.KafkaConfig
 import kafka.utils.TestUtils
 import org.apache.kafka.clients.admin.NewTopic
 import org.apache.kafka.clients.consumer.ConsumerConfig
 import org.apache.kafka.clients.producer.{ProducerConfig, ProducerRecord}
-import org.junit.Assert._
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.junit.runners.Parameterized
-import org.junit.runners.Parameterized.Parameters
+import org.junit.jupiter.api.Assertions._
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.{Arguments, MethodSource}
+
+import java.lang.{Boolean => JBoolean}
+import java.time.Duration
+import java.util
+import java.util.Collections
 
 /**
  * Tests behavior of specifying auto topic creation configuration for the consumer and broker
  */
-@RunWith(value = classOf[Parameterized])
-class ConsumerTopicCreationTest(brokerAutoTopicCreationEnable: JBoolean, consumerAllowAutoCreateTopics: JBoolean) extends IntegrationTestHarness {
+class ConsumerTopicCreationTest extends IntegrationTestHarness {
   override protected def brokerCount: Int = 1
 
   val topic_1 = "topic-1"
   val topic_2 = "topic-2"
   val producerClientId = "ConsumerTestProducer"
   val consumerClientId = "ConsumerTestConsumer"
 
-  // configure server properties
-  this.serverConfig.setProperty(KafkaConfig.ControlledShutdownEnableProp, "false") // speed up shutdown
-  this.serverConfig.setProperty(KafkaConfig.AutoCreateTopicsEnableProp, brokerAutoTopicCreationEnable.toString)
+  @BeforeEach
+  override def setUp(): Unit = {
+    // junit 5 can't make parametered BeforeEach so we override setup to do nothing as we do setup cluster in test case
+  }
```

Comment:
This is not ideal. Since we have a single test, maybe we can do the parameterization manually within the test.

Reply:
That requires to modify server configs "before" the ```@BeforeEach```. It seem to me the only way to address it is to separate the test class. WDYT?

Reply:
I was thinking we could use dynamic tests instead: https://www.baeldung.com/junit5-dynamic-tests#improving-the-dynamictest-using-java-8-features

Reply:
The last commit rewrite the test by nested test and it seems to me that is more readable than dynamic test. WDYT?

Reply:
Not sure. I think I would prefer 4 hardcoded test methods calling a private method with two booleans. Seems like it does the job in a simpler way.

Reply:
Basically:
1. `testAutoCreationEnabled`
2. `testBrokerAutoCreationEnabled`
3. `testClientAutoCreationEnabled`
4. `testAutoCreationDisabled`